### DISCUSSION TITLE
Wie relevant ist die Unterstützung von Homegear?

### DISCUSSION THREAD

**Discussion by SukramJ** [SukramJ/aiohomematic#30] [General]
Offene Frage. Ich habe keine Ahnung von Homegear!

Aus meiner Perspektive entstehen mit der Unterstützung von Homegear mehrere Unwägbarkeiten:
- teilweise doppelte Methoden, da Homegear z.Bsp. keine JSON-RPC unterstützt.
- Unterstützt der Homegear XMLXPC-Server system.multicall? Dadurch ließen sich Massenabfrage an die CCU zusammenfassen. Ob di CCU wirklich multicall unterstützt muss ich noch mal validieren.
- Wie ist das Verhalten bei mehreren parallel Zugriffen? Verhält sich Homegear anders als eine CCU?
- Ich verfüge über keine Homegear-Installation und wollte mich da auch nicht näher einarbeiten.

**Comment by danielperna84** [comment]
Also über die Jahre hinweg gab es schon nicht wenige Leute die im Forum Hilfe für das Setup mit Homegear gesucht haben. Ob die das immer noch benutzen weiß ich nicht. Aber ich würde es definitiv nicht einfach weg lassen. Homegear ist auch ziemlich gut programmiert und eigentlich kaum anders als die CCU ([multicall geht auch](https://ref.homegear.eu/rpc.html#systemmulticall)). Es fehlt halt nur das JSON-RPC, da hier die Namen stattdessen im Metadata gespeichert sind. Und die werden ja nicht ständig abgerufen. Da finde ich es also schon ok beides drin zu lassen.

Wenn du das im Code etwas ordentlicher haben willst kann man das natürlich auch umbauen damit es eher wie mit "Plugins" funktioniert. Und je nachdem welches Backend vorhanden ist wir halt das Plugin für die CCU oder eben für Homegear beim start initialisiert. Find' ich halt etwas zu viel Aufwand, da es vermutlich nicht mehr als diese beiden Wege gibt. Kann man also meiner Meinung nach einfach so lassen.

Was parallele Zugriffe angeht dürfte Homegear sogar noch stabiler sein als eine CCU. Vielleicht grob erklärt: Homegear kann verschiedene Systeme (Philips Hue, HomeMatic, EnOcean usw.) zusammenfassen und stellt alles per XML-RPC zur Verfügung. Ein bisschen so wie der cuxd, aber halt extern. Es ist also dafür konzipiert mehr oder minder ausschließlich über XML-RPC zu arbeiten. Bei der CCU ist die Schnittstelle hingegen ja eher ein Nebenfeature.

Also zusammengefasst: der XML-RPC Kram wird mit Homegear genauso funktionieren, und der Code für die Namen ist ja schon da.

**Reply by SukramJ** [reply]
Danke für die Einordnung. Das war hilfreich.
